src/compas_blender/artists/robotmodelartist.py

```diff
@@ -18,14 +23,29 @@ class RobotModelArtist(BaseRobotModelArtist):
         Robot model.
     """
 
-    def __init__(self, model, collection=None):
-        self.collection = collection
+    def __init__(self,
+                 model: RobotModel,
+                 collection: bpy.types.Collection = None):
+        self.collection = collection or model.name
         super(RobotModelArtist, self).__init__(model)
 
-    def transform(self, native_mesh, transformation):
+    @property
+    def collection(self) -> bpy.types.Collection:
+        return self._collection
+
+    @collection.setter
+    def collection(self, value: Union[str, bpy.types.Collection]):
+        if isinstance(value, bpy.types.Collection):
+            self._collection = value
+        elif isinstance(value, str):
+            self._collection = compas_blender.create_collection(value)
+        else:
+            raise Exception('Collection must be of type `str` or `bpy.types.Collection`.')
+
+    def transform(self, native_mesh: bpy.types.Object, transformation: Transformation) -> None:
         native_mesh.matrix_world = mathutils.Matrix(transformation.matrix) @ native_mesh.matrix_world
 
-    def create_geometry(self, geometry, name=None, color=None):
+    def create_geometry(self, geometry: Union[Mesh, Shape], name: str = None, color: Tuple = None) -> bpy.types.Object:
```

Comment:
`Union[Tuple[int, int, int], Tuple[float, float, float]]`? perhaps things like this can be defined in a central location...